vertical-pod-autoscaler/enhancements/4831-control-actuation-behavior/README.md

```diff
@@ -0,0 +1,106 @@
+# AEP-4831: Control VPA eviction behavior based on scaling direction and resource
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+   - [Goals](#goals)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+   - [Test Plan](#test-plan)
+- [Alternatives](#alternatives)
+- [Implementation History](#implementation-history)
+
+<!-- /toc -->
+
+## Summary
+VPA provides different `UpdateMode`s, describing if/how the VPA applies new recommendations to the Pods. They range from
+"recommendation only, don't make any changes" (`Off`), over "apply new recommendations when pods are re-scheduled, but
+don't evict" (`Initial`) to "evict a Pod to apply a new recommendation" (`Auto`). These existing `UpdateModes` work in
+the same way for scaling up and scaling down and are the same for all resources controlled by the VPA.
+
+[Kubernetes 1.27 supports](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.27.md#api-change-3)
+[changing container resources in place](https://github.com/kubernetes/enhancements/issues/1287) as an alpha feature.
+VPA will use that as
+[an alternative way to actuate its recommendations](https://github.com/kubernetes/autoscaler/issues/4016).
+
+We propose adding a new functionality which allows to control the conditions in which recommendations can be applied to
+containers based on the changes (are we scaling up or scaling down?) for each resource individually.
+
+## Motivation
+For some workloads, each eviction introduces disruptions for users. Examples include anything running as a singleton 
+(e.g. etcd) or workload that needs a lot of time to process data on startup (e.g. prometheus). Therefore, operators may
+want to limit the amount of evictions. We can do this by introducing a new tradeoff: Overspend on resources to increase 
+uptime. It is possible to turn off eviction for downscaling entirely or have it managed by an external component to 
+allow downscaling only during certain time windows, days, or based on other criteria.
+
+Scaling containers in place might cause immediate disruption (when containers specify `RestartPolicy: RestartContainer`
+for one of the changed resources) which operators might want to avoid. Even if in place scale down doesn't cause
+immediate disruption there is no guarantee that container will be able to scale up later without eviction. 
+
+### Goals
+* For a specific VPA object, allow turning off actuation for downscaling entirely while still allow for upscaling
+* Allow for resource-specific decisions: The desired policy may be different for CPU and Memory
```

Comment:
This might be open for other resource than CPU and Memory int he future, so let's pay attention to have an API open to handle other type of resources.

Reply:
I think the proposal is easy to extend to other resources. Do you expect any problems?